prometheus_juju_exporter/collector.py

```diff
@@ -77,14 +77,18 @@ async def _connect_controller(
                 endpoint=endpoint, username=username, password=password, cacert=cacert
             )
 
-    async def _get_machines_in_model(self, uuid: str) -> Dict[Any, Any]:
+    async def _get_machines_in_model(self, uuid: str) -> Optional[Dict[Any, Any]]:
         """Get a list of all machines in the model with their stats.
 
         :return: status information for all machines in the model
         """
-        model = await self.controller.get_model(uuid)
-        status = await model.get_status()
-        await model.disconnect()
+        try:
+            model = await self.controller.get_model(uuid)
+            status = await model.get_status()
+            await model.disconnect()
+        except Exception as err:  # pylint: disable=W0703
+            self.logger.error("Failed connecting to model '%s': %s ", uuid, err)
+            return None
```

Comment:
(non-blocker) Since you are not explicitly checking for `None` value anywhere, you could return just empty dict and wouldn't have to change method signature.

Reply:
Updated. Would appreciate another look from you.

Reply:
Thanks, I think this is bit cleaner approach.